src/lib/utils/application-commands/ApplicationCommandRegistries.ts

```diff
@@ -53,8 +58,119 @@ export async function handleRegistryAPICalls() {
+
+	// Collect all data
+	for (const command of commandStore.values()) {
+		const registry = command.applicationCommandRegistry;
+
+		for (const call of registry['apiCalls']) {
+			// Guild only cmd
+			if (call.registerOptions.guildIds?.length) {
+				for (const guildId of call.registerOptions.guildIds) {
+					foundGuildCommands[guildId] ??= [];
+
+					foundGuildCommands[guildId].push({ piece: command, data: call.builtData });
+				}
+				continue;
+			}
+
+			// Global command
+			foundGlobalCommands.push({ piece: command, data: call.builtData });
+		}
+	}
+
+	// Handle global commands
+	try {
+		bulkOverwriteDebug(`Overwriting global application commands, now at ${foundGlobalCommands.length} commands`);
+		const result = await applicationCommands.set(foundGlobalCommands.map((x) => x.data));
+
+		// Go through each registered command, find its piece and alias it
+		for (const [id, globalCommand] of result.entries()) {
+			// I really hope nobody has a command with the same name as another command -.-
+			// Not like they could anyways as Discord would throw an error for duplicate names
+			// But yknow... If you're reading this and you did this... Why?
+			const piece = foundGlobalCommands.find((x) => x.data.name === globalCommand.name)?.piece;
+
+			if (piece) {
+				const registry = piece.applicationCommandRegistry;
+
+				registry.globalCommandId = id;
+				registry.addChatInputCommandIds(id);
 
+				// idHints are useless, and any manually added id or names could end up not being valid anymore if you use bulk overwrites
+				// That said, this might be an issue, so we might need to do it like `handleAppendOrUpdate`
+				commandStore.aliases.set(id, piece);
+			} else {
+				bulkOverwriteWarn(
+					`Registered global command "${globalCommand.name}" (${id}) but failed to find the piece in the command store. This should not happen`
+				);
+			}
+		}
+
+		bulkOverwriteInfo(`Successfully overwrote global application commands. The application now has ${result.size} global commands`);
+	} catch (error) {
+		bulkOverwriteError(`Failed to overwrite global application commands`, error);
+	}
+
+	// Handle guild commands
+	for (const [guildId, guildCommands] of Object.entries(foundGuildCommands)) {
+		try {
+			bulkOverwriteDebug(`Overwriting guild application commands for guild ${guildId}, now at ${guildCommands.length} commands`);
+			const result = await applicationCommands.set(
+				guildCommands.map((x) => x.data),
+				guildId
+			);
+
+			// Go through each registered command, find its piece and alias it
+			for (const [id, guildCommand] of result.entries()) {
+				// I really hope nobody has a guild command with the same name as another command -.-
+				// Not like they could anyways as Discord would throw an error for duplicate names
+				// But yknow... If you're reading this and you did this... Why?
+				const piece = guildCommands.find((x) => x.data.name === guildCommand.name)?.piece;
```

Comment:
```suggestion
				const piece = guildCommands.find((x) => x.data.name === guildCommand.name)?.piece;
```

Unnecessary code comment, code smell. [Article one](https://techbeacon.com/app-dev-testing/useless-comments-can-ruin-your-code-reviews-heres-how-erase-them), [Article two](https://medium.com/@bpnorlander/stop-writing-code-comments-28fef5272752)